src/single-post.ejs

```diff
@@ -0,0 +1,622 @@
+        </button>
+        <button
+          class="page-footer__button page-footer__button--vertical footer-button"
+        >
+          <i class="footer-button__logo">
+            <img src="<%=require( "./icons/svgs/plus-circle-o.svg" ).default%>"
+            />
+          </i>
+          <span class="footer-button__title"> ثبت آگهی </span>
+        </button>
+        <button
+          class="page-footer__button page-footer__button--vertical footer-button"
+        >
+          <i class="footer-button__logo">
+            <img src="<%=require( "./icons/svgs/chat-o.svg" ).default%>" />
+          </i>
+          <span class="footer-button__title"> چت </span>
+        </button>
+        <button
+          class="page-footer__button page-footer__button--vertical footer-button"
+        >
+          <i class="footer-button__logo">
+            <img src="<%=require( "./icons/svgs/account-o.svg" ).default%>" />
+          </i>
+          <span class="footer-button__title"> دیوار من </span>
+        </button>
+      </footer>
+    </div>
+    <script defer async>
+      const url_string = window.location.href;
+      const url = new URL(url_string);
+      const token = url.searchParams.get("token");
+
+      const getInfo = async (token) => {
+        const data = await fetch(
+          `https://api.divar.ir/v8/posts-v2/web/${token}`,
+          {
+            headers: {
+              accept: "application/json-filled",
+              "accept-language": "en-US,en;q=0.9",
+              "sec-ch-ua":
+                '".Not/A)Brand";v="99", "Google Chrome";v="103", "Chromium";v="103"',
+              "sec-ch-ua-mobile": "?0",
+              "sec-ch-ua-platform": '"Linux"',
+              "sec-fetch-dest": "empty",
+              "sec-fetch-mode": "cors",
+              "sec-fetch-site": "same-site",
+            },
+            referrer: "https://divar.ir/",
+            referrerPolicy: "origin",
+            body: null,
+            method: "GET",
+            mode: "cors",
+            credentials: "omit",
+          }
+        )
+          .then((res) => res.json())
+          .then((res) => res);
+        return data;
+      };
+
+      getInfo(token)
+        .then((res) => {
+          const titleSection = res.sections.find(
+            (el) => el.section_name === "TITLE"
+          ).widgets[0].data;
+
+          const listData = res.sections.find(
+            (el) => el.section_name === "LIST_DATA"
+          );
+
+          const rowsData = listData.widgets
+            .filter((el) => el.widget_type === "UNEXPANDABLE_ROW")
+            .map((el) => el.data);
+
+          const groupInfoTitle = listData.widgets.find(
+            (el) => el.widget_type === "SECTION_TITLE_ROW"
+          )?.data?.title;
+
+          const groupInfoRow = listData.widgets.find(
+            (el) => el.widget_type === "GROUP_INFO_ROW"
+          )?.data?.items;
+
+          const featureInfo = listData.widgets.find(
```

Comment:
This functionality has been repeated several times and could've been abstracted. Check comments on Mahdi's MR